.github/workflows/testBuildAndPublish.yml

```diff
@@ -0,0 +1,143 @@
+# GitHub Actions Workflow created for testing and publishing the plugin in following steps:
+# - validate Gradle Wrapper,
+# - run test and verifyPlugin tasks,
+# - run buildPlugin task and prepare artifact for the further tests,
+# - run IntelliJ Plugin Verifier,
+# - publish plugin if the workflow was triggered by a release
+
+# Docs:
+# - GitHub Actions: https://help.github.com/en/actions
+# - IntelliJ Plugin Verifier GitHub Action: https://github.com/ChrisCarini/intellij-platform-plugin-verifier-action
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+  release:
+    types: [released]
```

Comment:
I wanted to ask if  the doc incomplete, 
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onevent_nametypes
but then I saw that details are here
https://docs.github.com/en/actions/reference/events-that-trigger-workflows#webhook-events
cool that you find out such stuff ...